crates/bevy_ecs/src/schedule/system_container.rs

```diff
@@ -9,7 +9,7 @@ use crate::{
 };
 use std::{borrow::Cow, ptr::NonNull};
 
-pub(super) trait SystemContainer {
+pub trait SystemContainer {
```

Comment:
This wasn't intended to be exposed, but I suppose it won't be a problem, as long as there's no mutable access. Ideally, it should be documented as well.

Reply:
The only thing that's mutable is `set_dependencies`, but the dependencies are only used for ambiguity checking so exposing that at least isn't unsound.

Reply:
We could add `#[doc(hidden)]` to some methods as well if we don't want to expose them.

Reply:
`.dependencies()` is used for ordering, too. It would indeed be best annotate it and `.set_dependencies()` with `#[doc(hidden)]`: it's an implementation detail that just doesn't make sense to external users.